ui/apps/platform/src/Containers/DeferralConfiguration/VulnerabilitiesConfiguration.tsx

```diff
@@ -9,54 +10,148 @@ import {
                 </FormGroup>
             </GridItem>
             <GridItem span={4} md={8} xl={9}>
                 <FormGroup>
-                    <Switch label="Enabled" labelOff="Disabled" isChecked={isEnabled} />
+                    <Switch
+                        id={`${fieldId}.enabled`}
+                        label="Enabled"
+                        labelOff="Disabled"
+                        isChecked={isEnabled}
+                        onChange={(_, e) => handleChange(e)}
+                    />
                 </FormGroup>
             </GridItem>
         </>
     );
 }
 
-function BooleanSetting({ label, isEnabled }: BaseSettingProps & { label: string }) {
+function BooleanSetting({
+    fieldId,
+    label,
+    isEnabled,
+    handleChange,
+}: BaseSettingProps & {
+    label: string;
+}) {
     return (
         <>
             <GridItem className="pf-u-py-xs" span={8} md={4} xl={3}>
                 <p>{label}</p>
             </GridItem>
             <GridItem className="pf-u-py-xs" span={4} md={8} xl={9}>
                 <FormGroup>
-                    <Switch label="Enabled" labelOff="Disabled" isChecked={isEnabled} />
+                    <Switch
+                        id={fieldId}
+                        label="Enabled"
+                        labelOff="Disabled"
+                        isChecked={isEnabled}
+                        onChange={(_, e) => handleChange(e)}
+                    />
                 </FormGroup>
             </GridItem>
         </>
     );
 }
+function getDefaultConfig(): VulnerabilitiesDeferralConfig {
+    return {
+        expiryOptions: {
+            dayOptions: [
+                { numDays: 1, enabled: false },
+                { numDays: 1, enabled: false },
+                { numDays: 1, enabled: false },
+                { numDays: 1, enabled: false },
+            ],
+            fixableCveOptions: {
+                allFixable: false,
+                anyFixable: false,
+            },
+            customDate: false,
+        },
+    };
+}
+
+function ensureMinimumDayOptions(
+    config: VulnerabilitiesDeferralConfig
+): VulnerabilitiesDeferralConfig {
+    const minimumLength = 4;
+    const dayOptions = [...config.expiryOptions.dayOptions];
+    while (dayOptions.length < minimumLength) {
+        dayOptions.push({ numDays: 1, enabled: false });
+    }
+
+    return {
+        ...config,
+        expiryOptions: {
+            ...config.expiryOptions,
+            dayOptions: sortBy(dayOptions, 'numDays'),
+        },
+    };
+}
 
 function VulnerabilitiesConfiguration() {
+    const { config, isConfigLoading, isUpdateInProgress, configLoadError } =
+        useVulnerabilitiesDeferralConfig();
+
+    const { values, handleChange } = useFormik({
+        enableReinitialize: true,
+        // Ensure that there are at least 4 day options in case this array was set to zero via the API
+        initialValues: ensureMinimumDayOptions(config ?? getDefaultConfig()),
```

Comment:
@dvail, quick question: Why did we opt for a minimum of 4-day options again? Instead of auto-setting it, wouldn't it be clearer to display an error if someone tries to save without them? Setting it to 0 and still seeing options seems a bit confusing.

Reply:
This is something that might be useful to bring up during a sync (or maybe just on Slack). I went with this option so that if zero options were saved via API, there would still be a way to reset them via the UI later. Saving zero or more than four options is supported so this was a way to normalize the UI with the mocks.

Although it probably would be simpler to just display exactly what the API returns, since this is likely a rare edge case, I thought I would give the user a fallback in case something went wrong in an API call at some point.